ra/ra.go

```diff
@@ -2122,6 +2139,21 @@ func (ra *RegistrationAuthorityImpl) NewOrder(ctx context.Context, req *rapb.New
 			"Order cannot contain more than %d DNS names", ra.maxNames)
 	}
 
+	if req.CertificateProfileName != "" && ra.validationProfiles != nil {
+		vp, ok := ra.validationProfiles[req.CertificateProfileName]
+		if !ok {
+			return nil, berrors.MalformedError("requested certificate profile %q not found",
+				req.CertificateProfileName,
+			)
+		}
+		if ok && vp.allowList != nil && !vp.allowList.Contains(req.RegistrationID) {
```

Comment:
The `if !ok` case above early-returns so we're guaranteed that `ok` is `true` by the time we get here.
```suggestion
		if vp.allowList != nil && !vp.allowList.Contains(req.RegistrationID) {
```